plugin/trino-hive/src/test/java/io/trino/plugin/hive/TestingHadoopServer.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.hive;
+
+import com.google.common.collect.ImmutableList;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+import org.testcontainers.containers.wait.strategy.HostPortWaitStrategy;
+import org.testcontainers.utility.DockerImageName;
+
+import java.io.Closeable;
+
+import static java.lang.String.format;
+
+public class TestingHadoopServer
+        implements Closeable
+{
+    private static final String HOSTNAME = "hadoop-master";
+
+    private final GenericContainer<?> dockerContainer;
+
+    public TestingHadoopServer()
+    {
+        dockerContainer = new GenericContainer<>(DockerImageName.parse("ghcr.io/trinodb/testing/hdp3.1-hive:41"))
+                .withCreateContainerCmdModifier(cmd -> cmd.withHostName(HOSTNAME))
+                .withExposedPorts(1180, 9083)
+                .withStartupCheckStrategy(new IsRunningStartupCheckStrategy())
+                .waitingFor(new HostPortWaitStrategy());
+        dockerContainer.setPortBindings(ImmutableList.of("9000:9000"));
+        dockerContainer.start();
+    }
+
+    public String getSocksProxy()
+    {
+        return format("%s:%s", HOSTNAME, dockerContainer.getMappedPort(1180));
```

Comment:
Does this actually work when you do **not** have `hadoop-master` in `/etc/hosts`?

i would assume the address is resolved outside of the docker container, and so should result in name resolution error.

Maybe  `dockerContainer.getContainerIpAddress()`  should be used here?

Reply:
It doesn't work if we don't have `hadoop-master` in `/etc/hosts`.

`GenericContainer.getContainerIpAddress()` causes "Malformed reply from SOCKS server" 
Also, other places depend on the hostname in my understanding. For instance, schema & table's `location` contains `hadoop-master`.

Reply:
> It doesn't work if we don't have `hadoop-master` in `/etc/hosts`.

That's IMO a no-go, tests shouldn't require system modification like that.